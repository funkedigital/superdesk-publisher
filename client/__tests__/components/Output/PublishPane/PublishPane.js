import React from "react";
import PublishPane from "../../../../components/Output/PublishPane/PublishPane";
import { render } from "@testing-library/react";
import Store from "../../../../components/Output/Store";

describe("Output/PublishPane/PublishPane", () => {

  it("renders correctly", () => {
    const { container } = render(
      <Store.Provider
        value={{
          publisher: { queryRelatedArticlesStatus: jest.fn().mockResolvedValue({ related_article_items: [] }) },
          selectedItem: { id: 1 },
          tenants: [
            {
              name: 'tenant1', code: 'tenant1', paywall_enabled: true, fbia_enabled: false,
              routes: [
                { name: 'testroute1', id: 1 },
                { name: 'testroute2', id: 2 },
              ]
            },
            {
              name: 'tenant2', code: 'tenant2', paywall_enabled: true, fbia_enabled: false,
              routes: [
                { name: 'testroute1', id: 1 },
                { name: 'testroute2', id: 2 },
              ],
              content_lists: []
            }
          ]
        }}
      >
        <PublishPane isOpen={true} />
      </Store.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();

  });

});
