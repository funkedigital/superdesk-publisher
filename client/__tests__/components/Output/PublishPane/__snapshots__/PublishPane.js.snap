// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/PublishPane/PublishPane renders correctly 1`] = `
<div
  class="sd-publish-panel"
>
  <div
    class="side-panel side-panel--shadow-right side-panel--dark-ui"
  >
    <div
      class="side-panel__header"
    >
      <div
        class="side-panel__tools"
      >
        <a
          class="icn-btn"
        >
          <i
            class="icon-close-small"
          />
        </a>
      </div>
      <ul
        class="nav-tabs nav-tabs--ui-dark"
      >
        <li
          class="nav-tabs__tab nav-tabs__tab--active"
        >
          <button
            class="nav-tabs__link"
          >
            <span>
              Publish
            </span>
          </button>
        </li>
        <li
          class="nav-tabs__tab"
        >
          <button
            class="nav-tabs__link"
          >
            <span>
              Unpublish
            </span>
          </button>
        </li>
      </ul>
    </div>
    <div
      class="side-panel__content"
    >
      <div
        class="side-panel__content-block side-panel__content-block--pad-small"
      >
        <div
          class="side-panel__content-block-heading side-panel__content-block-heading--small-margin"
        />
      </div>
      <div
        class="side-panel__content-block side-panel__content-block--flex side-panel__content-block--space-between"
      >
        <div
          class="dropdown undefined"
        >
          <span>
            <button
              class="btn btn--primary btn--icon-only-circle btn--large dropdown__toggle"
              flow="right"
              sd-tooltip="Add destination"
            >
              <i
                class="icon-plus-large"
              />
            </button>
          </span>
          <ul
            class="dropdown__menu dropdown__menu--scrollable"
          >
            <li>
              <button>
                tenant1
              </button>
            </li>
            <li>
              <button>
                tenant2
              </button>
            </li>
          </ul>
        </div>
        <div>
          <span
            class="sd-check__wrapper"
          >
            <span
              class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio checked"
            >
              <label>
                All
              </label>
            </span>
          </span>
          <span
            class="sd-check__wrapper"
          >
            <span
              class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio"
            >
              <label>
                Published
              </label>
            </span>
          </span>
          <span
            class="sd-check__wrapper"
          >
            <span
              class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio"
            >
              <label>
                Unpublished
              </label>
            </span>
          </span>
        </div>
      </div>
      <div
        class="side-panel__content-block side-panel__content-block--pad-small"
      />
      <div
        class="side-panel__content-block"
      >
        <div
          class="toggle-box toggle-box--dark sp--dark-ui"
        >
          <div
            class="toggle-box__header"
          >
            <div
              class="toggle-box__chevron"
            >
              <i
                class="icon-chevron-right-thin"
              />
            </div>
            <div
              class="toggle-box__label"
            >
              Related Articles
            </div>
            <div
              class="toggle-box__line"
            />
          </div>
          <div
            class="toggle-box__content-wraper"
          >
            <div
              class="toggle-box__content"
            >
              <div
                class="spinner"
              >
                <div
                  class="bounce1"
                />
                <div
                  class="bounce2"
                />
                <div
                  class="bounce3"
                />
              </div>
              <ul
                class="simple-list simple-list--dotted simple-list--no-padding"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="side-panel__content-block-overlay-grid"
    >
      <div
        class="side-panel"
      >
        <div
          class="side-panel__header"
        >
          <a
            class="icn-btn sd-margin-l--1"
            flow="right"
            sd-tooltip="Back"
          >
            <i
              class="icon-arrow-left"
            />
          </a>
          <h3
            class="side-panel__heading side-panel__heading--big"
          >
            facebook
             meta data
          </h3>
        </div>
        <div
          class="side-panel__content"
        >
          <div
            class="side-panel__content-block"
          >
            <div
              class="form__row"
            >
              <div
                class="sd-line-input"
              >
                <label
                  class="sd-line-input__label"
                >
                  Post image
                </label>
                <div
                  class="sd-overlay-block sd-overlay-block--no-image"
                >
                  <div
                    class="sd-overlay-block__overlay"
                  >
                    <div
                      class="sd-overlay-block__overlay-action-group"
                    >
                      <label
                        class="sd-overlay-block__overlay-action"
                        for="meta_media_file"
                        sd-tooltip="Upload image"
                      >
                        <i
                          class="icon-upload"
                        />
                      </label>
                      <span
                        class="sd-overlay-block__overlay-message"
                      >
                        No image has been set so far.
                      </span>
                    </div>
                  </div>
                </div>
                <input
                  accept="image/*"
                  id="meta_media_file"
                  name="meta_media_file"
                  style="opacity: 0; position: absolute; width: 1px; height: 1px;"
                  type="file"
                />
              </div>
            </div>
            <div
              class="form__row"
            >
              <div
                class="sd-line-input sd-line-input--dark-ui sd-line-input--boxed"
              >
                <label
                  class="sd-line-input__label"
                >
                  Title
                </label>
                <input
                  class="sd-line-input__input"
                  name="title"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form__row"
            >
              <div
                class="sd-line-input sd-line-input--dark-ui sd-line-input--boxed"
              >
                <label
                  class="sd-line-input__label"
                >
                  Description
                </label>
                <textarea
                  class="sd-line-input__input"
                  name="description"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="side-panel__footer side-panel__footer--button-box-large"
    >
      <button
        class="btn btn--large btn--success btn--expanded"
        disabled=""
      >
        Publish
      </button>
    </div>
  </div>
</div>
`;
